plugins/com.google.cloud.tools.eclipse.appengine.ui.test/src/com/google/cloud/tools/eclipse/appengine/ui/AppEngineLibrariesSelectorGroupTest.java

```diff
@@ -0,0 +1,256 @@
+/*
+7 * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.ui;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import com.google.cloud.tools.eclipse.appengine.libraries.model.Library;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import org.eclipse.swt.layout.FillLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotCheckBox;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class AppEngineLibrariesSelectorGroupTest {
+
+  private Shell shell;
+  private AppEngineLibrariesSelectorGroup librariesSelector;
+  private SWTBotCheckBox appengineButton;
+  private SWTBotCheckBox endpointsButton;
+  private SWTBotCheckBox objectifyButton;
+
+  @Before
+  public void setUp() throws Exception {
+    Display.getDefault().syncExec(new Runnable() {
+
+      @Override
+      public void run() {
+        shell = new Shell(Display.getDefault());
+        shell.setLayout(new FillLayout());
+        librariesSelector = new AppEngineLibrariesSelectorGroup(shell);
+        shell.open();
+        appengineButton = getButton("appengine-api");
+        endpointsButton = getButton("appengine-endpoints");
+        objectifyButton = getButton("objectify");
+      }
+    });
+  }
+
+  @After
+  public void tearDown() {
+    Display.getDefault().syncExec(new Runnable() {
+
+      @Override
+      public void run() {
+        if (shell != null) {
```

Comment:
I prefer to avoid NPEs as I cannot always tell easily whether it's intended and OK or programming error. Also, we already throw so many ignored exceptions during tests that I'd rather not add to the noise.